android/src/main/java/com/hypertrack/sdk/capacitor/CapacitorSerialization.kt

```
@@ -0,0 +1,138 @@
+
+@Suppress("UNCHECKED_CAST")
+internal fun <T> Result<T>.toPluginCall(call: PluginCall) {
+  when(this) {
+    is Success -> {
+      when(this.success) {
+        is Map<*, *> -> {
+            call.resolve((this.success as Map<String, Any>).toJSObject())
+        }
+        is Unit -> {
+            call.resolve()
+        }
+        else -> {
+            val exception = IllegalArgumentException(this.success.toString())
+            call.reject(exception.message, exception)
+        }
+      }
+    }
+    is Failure -> {
+        val exception = this.failure
+        if(exception is Exception) {
+            call.reject(exception.message, exception)
+        } else {
+            throw exception
+        }
+    }
+  }
+}
+
+@Suppress("UNCHECKED_CAST")
+internal fun List<Any>.toJSArray(): JSArray {
+    return JSArray().also { writableArray ->
+        forEach {
+            when (it) {
+                is String -> {
+                    writableArray.put(it)
+                }
+                is Map<*, *> -> {
+                    writableArray.put((it as Map<String, Any>).toJSObject())
+                }
+                else -> {
+                    throw Exception(IllegalArgumentException(it.javaClass.toString()))
+                }
+            }
+        }
+    }
+}
+
+@Suppress("UNCHECKED_CAST")
+internal fun Map<String, Any?>.toJSObject(): JSObject {
+    val map = this
+    return JSObject().apply {
+        map.forEach { entry ->
+            val key = entry.key
+            when (val value = entry.value) {
+                is String -> {
+                    put(key, value)
+                }
+                is Int -> {
+                    put(key, value)
+                }
+                is Double -> {
+                    put(key, value)
+                }
+                is Boolean -> {
+                    put(key, value)
+                }
+                is List<*> -> {
+                    put(key, (value as List<String>).toJSArray())
```

I don't understand why the check is for a wildcard and then the forced cast is to `List<String>` It should be both the same or it may crash.

---

AFAIK we can't check for generic in List without reflection, and non-string list is imposible in this code

---

Solved on the call with Eugene (decision: to fail fast and allow crashes on wrong data)